File: src/bot/RobotPlayer.scala
package bot

import battlecode.common._

object RobotPlayer {
	/**
	  * run() is the method that is called when a robot is instantiated in the Battlecode world.
	  * If this method returns, the robot dies!
	  **/
	@SuppressWarnings(Array("unused"))
	@throws[GameActionException]
	def run(rc: RobotController) {
		// Here, we've separated the controls into a different method for each RobotType.
		try {
			val robot = rc.getType match {
				case RobotType.ARCHON => new Archon()
				case RobotType.GARDENER => new Gardener()
				case RobotType.SOLDIER => new Soldier()
				case RobotType.LUMBERJACK => new Lumberjack()
				case RobotType.SCOUT => new Scout()
				case _ => return
			}

			robot.rc = rc
			robot.init()
			robot.run()
		} catch {
			case e: Exception => {
				System.out.println("Exception in " + rc.getType)
				e.printStackTrace()

				// Initialize the unit completely from scratch
				//run(rc)
			}
		}
	}
}

File: src/bot/RobotPlayer.java
package bot;

import battlecode.common.*;

@SuppressWarnings("unused")
public strictfp class RobotPlayer {

    @SuppressWarnings("unused")
    public static void run(RobotController rc) {
        Robot robot = null;
        try {
            Robot.init(rc);
            switch (rc.getType()) {
                case ARCHON:
                    robot = new Archon();
                    break;
                case GARDENER:
                    robot = new Gardener();
                    break;
                case SOLDIER:
                    robot = new Soldier();
                    break;
                case LUMBERJACK:
                    robot = new Lumberjack();
                    break;
                case SCOUT:
                    robot = new Scout();
                    break;
                case TANK:
                    robot = new Tank();
                    break;
            }

            robot.onAwake();
            robot.onStartOfTick();
        } catch (Exception e) {
            System.out.println("Exception in " + rc.getType());
            e.printStackTrace();
        }

        //noinspection InfiniteLoopStatement
        while (true) {
            try {
                //noinspection InfiniteLoopStatement
                while(true) {
                    robot.onUpdate();
                }
            } catch (Exception e) {
                System.out.println("Exception in " + rc.getType());
                e.printStackTrace();
            }
        }
    }
}
